app/scripts/modules/core/delivery/service/execution.service.js

```diff
@@ -260,6 +260,18 @@ module.exports = angular.module('spinnaker.core.delivery.executions.service', [
       }
     }
 
+    function getLastExecution(appName, configId) {
+      return getExecutions(appName)
```

Comment:
If you change this to `getFilteredExecutions(appName, {}, 1)`, it should be quicker. Also probably just want to return the first item or null - the name implies that's what would happen.

Otherwise, might want to rename the method to `getExecutionsForConfigId` or something, since it's returning all executions (or whatever the `ExecutionFilterModel.sortFilter.count` is set to when the call happens).